src/components/post/post.service.ts

```diff
@@ -0,0 +1,313 @@
+      );
+    }
+
+    try {
+      const createPost = this.db
+        .query()
+        .call(matchRequestingUser, session)
+        .call(createBaseNode, postId, 'Post', secureProps)
+        .return('node.id as id');
+
+      await createPost.first();
+
+      // connect the Post to the User
+      await this.db
+        .query()
+        .match([
+          [node('user', 'User', { id: session.userId })],
+          [node('post', 'Post', { id: postId })],
+        ])
+        .create([
+          node('post'),
+          relation('out', '', 'user', {
+            active: true,
+            createdAt: DateTime.local(),
+          }),
+          node('user'),
+        ])
+        .return('post.id as id')
+        .first();
+
+      // Connect the post to the project, if applicable
+      if (projectId) {
+        await this.db
+          .query()
+          .match([
+            [node('project', 'Project', { id: projectId })],
+            [node('post', 'Post', { id: postId })],
+          ])
+          .create([
+            node('project'),
+            relation('out', '', 'project', {
+              active: true,
+              createdAt: DateTime.local(),
+            }),
+            node('post'),
+          ])
+          .return('post.id as id')
+          .first();
+      }
+      // connect the post to the engagement, if applicable
+      else if (engagementId) {
+        await this.db
+          .query()
+          .match([
+            [node('engagement', 'Engagement', { id: engagementId })],
+            [node('post', 'Post', { id: postId })],
+          ])
+          .create([
+            node('engagement'),
+            relation('out', '', 'engagement', {
+              active: true,
+              createdAt: DateTime.local(),
+            }),
+            node('post'),
+          ])
+          .return('post.id as id')
+          .first();
+      }
+      // connect the post to a parent post, if applicable
+      else if (parentPostId) {
+        await this.db
+          .query()
+          .match([
+            [node('parentPost', 'Post', { id: parentPostId })],
+            [node('post', 'Post', { id: postId })],
+          ])
+          .create([
+            node('parentPost'),
+            relation('out', '', 'parentPost', {
+              active: true,
+              createdAt: DateTime.local(),
+            }),
+            node('post'),
+          ])
+          .return('post.id as id')
+          .first();
+      }
+
+      // FIXME: This is being refactored - leaving it commented out per Michael's instructions for now
+      // await this.authorizationService.processNewBaseNode(
+      //   new DbPost(),
+      //   postId,
+      //   session.userId
+      // );
+
+      return await this.readOne(postId, session);
+    } catch (exception) {
+      this.logger.warning('Failed to create post', {
+        exception,
+      });
+
+      throw new ServerException('Failed to create post', exception);
```

Comment:
Need to confirm that `parentId` is valid. If it's not, we need to throw `InputException` instead.